internal/index/index.go

```diff
@@ -317,30 +317,23 @@ type blobJSON struct {
 }
 
 // generatePackList returns a list of packs.
-func (idx *Index) generatePackList() ([]*packJSON, error) {
-	list := []*packJSON{}
-	packs := make(map[restic.ID]*packJSON)
```

Comment:
I'm somewhat unsure whether we should get rid of the map. Strictly speaking the index implementation does not guarantee a 1:1 mapping between pack IDs and the packIndex. If a pack is contained in multiple indexes, then after merging multiple indexes, the same pack ID might correspond to multiple packIndexes. However, `MasterIndex.Save`, which is the main user of `generatePackList`, that won't be a problem.

Reply:
What would be the benefit of just removing the debug.Log calls?

Reply:
```
name                 old time/op    new time/op    delta
EncodeIndex/100-8      1.56ms ± 2%    1.50ms ± 1%   -3.91%  (p=0.000 n=10+9)
EncodeIndex/1000-8     14.5ms ± 2%    13.3ms ± 2%   -8.38%  (p=0.000 n=9+9)
EncodeIndex/10000-8     120ms ± 2%     116ms ± 2%   -3.66%  (p=0.000 n=10+9)

name                 old alloc/op   new alloc/op   delta
EncodeIndex/100-8       306kB ± 1%     276kB ± 1%   -9.91%  (p=0.000 n=10+10)
EncodeIndex/1000-8     3.69MB ±11%    2.98MB ± 5%  -19.19%  (p=0.000 n=10+10)
EncodeIndex/10000-8    35.9MB ±11%    30.3MB ±17%  -15.58%  (p=0.000 n=10+10)

name                 old allocs/op  new allocs/op  delta
EncodeIndex/100-8       3.39k ± 0%     2.48k ± 0%  -27.05%  (p=0.000 n=10+10)
EncodeIndex/1000-8      32.6k ± 0%     23.9k ± 0%  -26.73%  (p=0.000 n=10+10)
EncodeIndex/10000-8      326k ± 0%      239k ± 0%  -26.73%  (p=0.000 n=10+10)
```

But there's an intermediate option, where the list still changes from []*packJSON to []packJSON. Let me try that.